clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -382,8 +382,24 @@ public static MemoryRecordsBuilder builder(ByteBuffer buffer,
                                                int baseSequence,
                                                boolean isTransactional,
                                                int partitionLeaderEpoch) {
+        return builder(buffer, magic, compressionType, timestampType, baseOffset,
+                logAppendTime, producerId, producerEpoch, baseSequence, isTransactional, false, partitionLeaderEpoch);
+    }
+
+    public static MemoryRecordsBuilder builder(ByteBuffer buffer,
```

Comment:
This method passes in `buffer.remaining()` as an additional parameter to the `MemoryRecordsBuilder` constructor.